File: php/sort/src/BubbleSort/BubbleSort.php
<?php

namespace Sort\BubbleSort;

use Sort\Support\UtilTrait;

class BubbleSort
{
    use UtilTrait;

    /**
     * 冒泡排序。
     *
     * @param array $array 待排序数组
     *
     * @return array
     */
    public static function sort(array $array): array
    {
        $count = count($array);
        if ($count < 2) {
            return $array;
        }

        for ($i = 0; $i < $count; $i++) {
            for ($j = 0; $j < $count - 1 - $i; $j++) {
                if (self::compare($array[$j], $array[$j + 1]) > 0) {
                    self::swap($array, $j, $j + 1);
                }
            }
        }

        return $array;
    }
}

File: java/sort/src/main/java/sort/bubble/BubbleSort.java
package sort.bubble;

import sort.util.SortUtil;

public class BubbleSort {

    /**
     * 冒泡排序。
     *
     * @param array 待排序数组
     *
     * @return 返回排序数组
     */
    public static int[] sort(int[] array) {
        if (array.length < 2) {
            return array;
        }

        for (int i = 0; i < array.length; i++) {
            for (int j = 0; j < array.length - 1 - i; j++) {
                if (SortUtil.compare(array[j], array[j + 1])) {
                    SortUtil.swap(array, j, j + 1);
                }
            }
        }

        return array;
    }

}
